src/instructlab/model/llmaaj.py

```diff
@@ -0,0 +1,164 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# pylint: disable=ungrouped-imports
+# Standard
+from datetime import datetime
+import enum
+import logging
+import pathlib
+import os
+
+# Third Party
+import httpx
+from openai import OpenAI, OpenAIError
+import pandas as pd
+from ragas.callbacks import ChainRun
+from ragas.dataset_schema import EvaluationDataset, EvaluationResult
+
+# First Party
+from instructlab.eval.ragas import ModelConfig, RagasEvaluator, RunConfig, Sample
+from instructlab.eval.mt_bench_common import get_openai_client
+from instructlab import client_utils
+
+# Local
+from ..utils import is_model_gguf, is_model_safetensors
+from ..client_utils import http_client, ClientException
+from .evaluate import launch_server, get_model_name
+from ..configuration import DEFAULTS
+
+logger = logging.getLogger(__name__)
+
+class IOFileType(enum.Enum):
+    CSV: str = "csv"
+    JSONL: str = "jsonl"
+    XLSX: str = "xlsx"
+    QNA_YAML: str = "qna.yaml"
+
+def model_is_local(model: str) -> bool:
+    if os.path.exists(model):
+        model_path = pathlib.Path(model)
+        valid_model = False
+        if model_path.is_dir():
+            valid_model = is_model_safetensors(model_path)
+        elif model_path.is_file():
+            valid_model = is_model_gguf(model_path)
+        return valid_model
+    else:
+        return False
+
+def get_endpoint_model_name(endpoint: str, api_key: str, http_client: httpx.Client) -> str:
+    try:
+        client = OpenAI(
+            base_url=endpoint,
+            api_key=api_key,
+            timeout=DEFAULTS.CONNECTION_TIMEOUT,
+            http_client=http_client,
+        )
+        models = client.models.list()
+    except OpenAIError as exc:
+        raise client_utils.ClientException(f"Connection Error {exc}") from exc
+
+    if len(models.data) != 1:
+        raise client_utils.ClientException(f"More than one model at endpoint, pass in model name manually")
+
+    return models.data[0].id
+
+def create_results_file_name(file_format: IOFileType, output_dir: str) -> str:
+    file_type = "jsonl"
+    if IOFileType.CSV.value == file_format:
+        file_type = "csv"
+    if IOFileType.XLSX.value == file_format:
+        file_type = "xlsx"
+
+    now = datetime.now()
+    timestamp = now.strftime("%m-%d-%Y_%H-%M-%S")
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.EVAL_DATA_DIR
+        logger.info(f"Output dir provided is not a directory or does not exist, writing results to output directory {output_dir}")
+
+    return f"{output_dir}/responses-{timestamp}.{file_type}"
+
+def run_llmaaj(ctx, model, max_workers, gpus, backend, enable_serving_output, input_questions, output_file_formats, output_dir, model_prompt, temperature, model_name, judge_model_name):
+    logger.info(f"Input questions path is {input_questions}")
```

Comment:
We should avoid logging by interpolating into the string. We want to instead pass it in as an argument. This is because the logger needs to format everything internally, so we have to pass it in like the classic C style.  

```suggestion
    logger.info("Input questions path is %s", input_questions)
```